### ISSUE TITLE
Takes subset of a word and flags it as profane like ass in assessment

### ISSUE THREAD

**Issue by buzzwin** [roshanrahman/profanity_filter#9]
Two words that were rejected: practitioner * assessment. Reason? Because they contained letters in a row that named certain body parts, is there a way to prevent this from happening.

**Comment by Desync-o-tron** [comment]
This is huge..

**Comment by roshanrahman** [comment]
Hi @buzzwin, thanks for pointing that out.  Turns out the `getAllProfanity` method is poorly written. ([Link to the test that it fails](https://github.com/roshanrahman/profanity_filter/blob/f870a54f6ee32598c8680b223dc6d2b1e99427e7/test/profanity_filter_test.dart#L77))

I'll get it fixed shortly. 

In the meantime, if you only want to find if the string contains profanity or not, you can use the `hasProfanity()` method (it passes the substring test). To get a list of detected profanity, something like this might help:

```dart
if(filter.hasProfanity(yourString)) { 
var listOfProfanity = filter.getAllProfanity(yourString);
}
```

**Comment by roshanrahman** [comment]
Published version 2.0.0 of the package that contains a fix for this. 
https://pub.dev/packages/profanity_filter/versions/2.0.0

The test cases all pass, and it should now work correctly. Closing the issue for now. 

Feel free to open new issue if you encounter any bugs.
